### ISSUE TITLE
Big bug - last part of url are not translated to alternate language - the same value of rawRoute() is used for all languages

### ISSUE THREAD

**Issue by Maofree** [getgrav/grav-plugin-langswitcher#63]
Hi
use as example this website
http://www.cremonte.me/proizvodi

if you are in the english version of that website and pass the mouse hover the "Me" and "English" you can see that the urls are both in the english at the end part.

now if you change to "Me" language, the last part with be both in russian

http://www.cremonte.me/proizvodi
http://www.cremonte.me/en/proizvodi

http://www.cremonte.me/proizvodi
http://www.cremonte.me/en/products

looking inside langswitcher.php at the line 78 you pass the same value to all languages

`$data->page_route = $page->rawRoute();`

instead all urls should be different for each language

now I've seen that rawRoute use $this->folder instead of $this->slug

so the url of not current languages are wrong

After some hours I've solved all bugs in this plugin, I can help you to fix them to get a better version of grav

**Comment by Maofree** [comment]
I'm going to explain the problems inside this plugin

inside onTwigSiteVariables it is wrong to use the condition

`if ($this->config->get('plugins.langswitcher.untranslated_pages_behavior') !== 'none') {`

because it is necessary to fill the variable
$translated_pages
if not the last part of urls will be with the folder name

so to avoid this problem, I need to set the option "Show language, link to home route"

into langswitcher.html.twig it is necessary to use

`{% set lang_url = base_lang_url ~ '/' ~ langswitcher.translated_pages[language].slug ~ page.urlExtension %}`

and into langswitcher.hreflang.html.twig

`{% set lang_url = base_url_simple ~ (langobj.getLanguageURLPrefix(key) ~ '/' ~ langswitcher.translated_pages[key].slug ~ page.urlExtension ?: '')|rtrim('/') %}`

![1](https://user-images.githubusercontent.com/653230/167887832-ee9ac87e-75b8-4c28-b3a8-77476a49e68c.png)
![2](https://user-images.githubusercontent.com/653230/167887861-37a4b268-b72f-466e-8848-05eb6d66be16.png)

[files.zip](https://github.com/getgrav/grav-plugin-langswitcher/files/8671284/files.zip)

I've done some changes to add x-default

without these fixes all sites with multi-languages are with wrong urls in the switch select

bye

**Comment by nicutor** [comment]
Hi @maofree,

Can you find a fix to make this to work on a multisite setup?

I get the same result like on these issues: https://github.com/getgrav/grav-plugin-langswitcher/issues/26 and https://github.com/getgrav/grav-plugin-langswitcher/issues/62

The issues are very old and nobody replied to them.. 

Thank you!

**Comment by Maofree** [comment]
Hi @nicutor 
have you test my solution on your site?
I need a multisite to test it

**Comment by nicutor** [comment]
@maofree yes, I've tested it. It work, but is not adding the multisite-setup prefix before /lang code.
Should be:

```
domain.com/product1/subpage
domain.com/product1/fr/subpage
```

but is:

```
domain.com/product1/subpage
domain.com/fr/subpage
```

**Comment by Maofree** [comment]
now I have too much work and I cannot free myself to do anything else, as soon as I can check and correct

**Comment by rhukster** [comment]
Please check the latest version of the develop branch (can download as zip).  I've completely rewritten this logic to use proper grav page urls, so it should use the expected link/url./

**Comment by Maofree** [comment]
Hi @rhukster 
I've tested your fix

there is a problem with langswitcher.hreflang.html.twig
it is missing the website domain from x-default and alternate link

![2](https://user-images.githubusercontent.com/653230/185599134-f5cba229-da14-4116-bde2-266e09dcf9b1.png)

langswitcher.html.twig is ok

**Comment by rhukster** [comment]
This is fixed in 3.0.0

**Comment by Maofree** [comment]
Hi @rhukster 
sorry but there is a new error in langswitcher.hreflang.html.twig
yes now there is the website domain but the urls are not correct

this happen when it is selected the english language
![1](https://user-images.githubusercontent.com/653230/185696388-f99b355d-ab89-4976-aeaa-f1b2ab3e4e2f.png)

here when it is set the default italian language
![2](https://user-images.githubusercontent.com/653230/185696358-6f79a64c-f37f-48a5-8a64-6e816eac713a.png)

look at the canonical url, the alternate urls should be similar to him
you should add just the domain to the previous solution

**Comment by rhukster** [comment]
Is this with the released  v3.0.0 of langswitcher?

**Comment by rhukster** [comment]
I found the solution, in line 12 of the hreflang.html.twig: 

```
{% if (language_obj.default == language and config.languages.include_default_lang == false) %}
```

**Comment by rhukster** [comment]
released as v3.0.1

**Comment by Maofree** [comment]
yes now all seems to be ok

thanks for your support
